### DISCUSSION TITLE
trouble operating audi with xbox controller disconnected from computer

### DISCUSSION THREAD

**Discussion by pruehr** [pybricks/support#1956] [Q&A]
I have have flashed my technique hub with pybricks and am able to connect to both the xbox controller and run the pybricks code.  Everything works fine.  When I turn both the controller and the hub off and then start the hub and the xbox controller, they will not connect to each other.  I have tried turning off the mac as I thought it may have been trying to connect with that.  I am powering on the controller prior to starting the hub.  I did go on the microsoft website with my pc and made sure my controller firmware was up to date.   I get a message that the computer is disconnecting so I assumed the code was downloaded to the hub.  Is there something else I need to do to get the pybricks code onto the hub?

**Comment by laurensvalk** [comment]
> When I turn both the controller and the hub off and then start the hub and the xbox controller, they will not connect to each other.

You have to turn on the hub before you can start the program, so that is two clicks in total.

Have a look at the light indications table here: https://pybricks.com/learn/getting-started/pybricks-environment/#start-programs-with-the-hub-button

When you start the program, the light should start to fade in and out.

If you want to be absolutely sure that your program is in fact running, it helps to add a command at the top of your code that changes the color of the hub light to red, for example.

**Reply by pruehr** [reply]
thanks so much, that worked!!!  Good idea for light.
